src/components/v5/common/FourOFourMessage/FourOFourMessage.tsx

```diff
@@ -0,0 +1,104 @@
+import { SmileySad } from '@phosphor-icons/react';
+import React, { type FC } from 'react';
+import { defineMessages } from 'react-intl';
+import { Link } from 'react-router-dom';
+
+import { openFeaturesBugs } from '~hooks/useBeamer.ts';
+import { formatText } from '~utils/intl.ts';
+import ButtonLink from '~v5/shared/Button/ButtonLink.tsx';
+import Button from '~v5/shared/Button/index.ts';
+
+import {
+  FourOFourMessageLinkType,
+  type FourOFourMessageProps,
+} from './types.ts';
+
+const displayName = 'v5.common.FourOFourMessage';
+
+const MSG = defineMessages({
+  title: {
+    id: `${displayName}.title`,
+    defaultMessage: 'Oops!',
+  },
+  errorCode: {
+    id: `${displayName}.errorCode`,
+    defaultMessage: 'Error code: 404',
+  },
+  reportBugBtn: {
+    id: `${displayName}.reportBug`,
+    defaultMessage: 'Report a bug',
+  },
+  linksTitle: {
+    id: `${displayName}.linksTitle`,
+    defaultMessage: 'Helpful links',
+  },
+});
+
+const FourOFourMessage: FC<FourOFourMessageProps> = ({
+  description,
+  links,
+  primaryLinkButton,
```

Comment:
could we maybe just compose these in? so we just render the `primaryLinkButton` as passed next to the report button.
The links are a bit trickier, but I think it makes sense to just compose them in and have the parent component control if the link is external or internal. It's a bit tricky to get the styles in (maybe we could wrap each of them in a div and apply the styles there, or try CSS selectors from this component).